collector/terminal_services.go

```diff
@@ -0,0 +1,403 @@
+	PercentUserTime             *prometheus.Desc
+	PoolNonpagedBytes           *prometheus.Desc
+	PoolPagedBytes              *prometheus.Desc
+	PrivateBytes                *prometheus.Desc
+	ThreadCount                 *prometheus.Desc
+	VirtualBytes                *prometheus.Desc
+	VirtualBytesPeak            *prometheus.Desc
+	WorkingSet                  *prometheus.Desc
+	WorkingSetPeak              *prometheus.Desc
+}
+
+// NewTerminalServicesCollector ...
+func NewTerminalServicesCollector() (Collector, error) {
+	const subsystem = "terminal_services"
+	return &TerminalServicesCollector{
+		LocalSessionCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "local_session_count"),
+			"Number of Terminal Services sessions",
+			[]string{"session"},
+			nil,
+		),
+		ConnectionBrokerPerformance: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "connection_broker_performance_total"),
+			"The total number of connections handled by the Connection Brokers since the service started.",
+			[]string{"connection"},
+			nil,
+		),
+		HandleCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "handle_count"),
+			"Total number of handles currently opened by this process. This number is the sum of the handles currently opened by each thread in this process.",
+			[]string{"session_name"},
+			nil,
+		),
+		PageFaultsPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "page_fault_per_sec"),
+			"Rate at which page faults occur in the threads executing in this process. A page fault occurs when a thread refers to a virtual memory page that is not in its working set in main memory. The page may not be retrieved from disk if it is on the standby list and therefore already in main memory. The page also may not be retrieved if it is in use by another process which shares the page.",
+			[]string{"session_name"},
+			nil,
+		),
+		PageFileBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "page_file_bytes"),
+			"Current number of bytes this process has used in the paging file(s). Paging files are used to store pages of memory used by the process that are not contained in other files. Paging files are shared by all processes, and lack of space in paging files can prevent other processes from allocating memory.",
+			[]string{"session_name"},
+			nil,
+		),
+		PageFileBytesPeak: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "page_file_bytes_peak"),
+			"Maximum number of bytes this process has used in the paging file(s). Paging files are used to store pages of memory used by the process that are not contained in other files. Paging files are shared by all processes, and lack of space in paging files can prevent other processes from allocating memory.",
+			[]string{"session_name"},
+			nil,
+		),
+		PercentPrivilegedTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "privileged_time_total"),
```

Comment:
Per the conventions, we should have the unit here as well
```suggestion
			prometheus.BuildFQName(Namespace, subsystem, "privileged_time_seconds_total"),
```